### PR TITLE
3rd Integration Test Update and Addition of Live Homing Rockets

### PR DESCRIPTION
**Pull request by aaronschneider7797**
Additions:
Live Homing Rockets that Target the Player Only.
Added Entries to Swiss German Language.

Changes:
Update to MCreator 2022.1 EAP 12711.
Prioritized Structure Spawn for Carminite Bases.

Fixes:
Fixed Version Overlay.

### CODE REVIEW

**Comment by PechayClub** [comment]
It was the `_fv` that has changed.

File: src/main/java/ch/teamburnuts/tnt_and_disparity/block/BBBBlock.java [hunk 1]
```diff
@@ -44,8 +44,8 @@
 
 			EntityBlock {
 	public BBBBlock() {
-		super(BlockBehaviour.Properties.of(Material.EXPLOSIVE).sound(SoundType.STONE).strength(80f, 11f).requiresCorrectToolForDrops().randomTicks()
```
Comment (PechayClub) [inline src/main/java/ch/teamburnuts/tnt_and_disparity/block/BBBBlock.java:47]: Change in MCreator version lead also to the change of the code as well.

File: elements/ArmorEffect.mod.json [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {
-  "_fv": 24,
```
Comment (PechayClub) [inline elements/ArmorEffect.mod.json:2]: Updating the MCreator version lead to the change of the `_fv` in the non code asset files

File: src/main/java/ch/teamburnuts/tnt_and_disparity/block/BBBBlock.java [hunk 3]
```diff
@@ -44,8 +44,8 @@
...
-				.hasPostProcess((bs, br, bp) -> true).emissiveRendering((bs, br, bp) -> true));
+		super(BlockBehaviour.Properties.of(Material.EXPLOSIVE, MaterialColor.FIRE).sound(SoundType.STONE).strength(80f, 11f)
+				.requiresCorrectToolForDrops().randomTicks().hasPostProcess((bs, br, bp) -> true).emissiveRendering((bs, br, bp) -> true));
```
Comment (PechayClub) [inline src/main/java/ch/teamburnuts/tnt_and_disparity/block/BBBBlock.java:48]: Updating the MCreator version lead to the change of the code.

File: src/main/java/ch/teamburnuts/tnt_and_disparity/block/BBBBlock.java [hunk 4]
```diff
@@ -65,11 +65,6 @@ public int getLightBlock(BlockState state, BlockGetter worldIn, BlockPos pos) {
 		return 0;
 	}
 
-	@Override
-	public MaterialColor defaultMaterialColor() {
-		return MaterialColor.FIRE;
-	}
-
```
Comment (PechayClub) [inline src/main/java/ch/teamburnuts/tnt_and_disparity/block/BBBBlock.java:72]: Updating the MCreator version lead to the change of the code.

File: src/main/java/ch/teamburnuts/tnt_and_disparity/block/DLKLandPortalBlock.java [hunk 5]
```diff
@@ -37,7 +37,7 @@ public DLKLandPortalBlock() {
 	}
 
 	@Override
-	public void tick(BlockState state, ServerLevel world, BlockPos pos, Random random) {
+	public void tick(BlockState blockstate, ServerLevel world, BlockPos pos, Random random) {
```
Comment (PechayClub) [inline src/main/java/ch/teamburnuts/tnt_and_disparity/block/DLKLandPortalBlock.java:40]: Updating the MCreator version lead to the change of the code.

File: src/main/java/ch/teamburnuts/tnt_and_disparity/block/GlowBlock.java [hunk 6]
```diff
@@ -28,8 +28,8 @@
 
 public class GlowBlock extends Block {
 	public GlowBlock() {
-		super(BlockBehaviour.Properties.of(Material.BARRIER).sound(SoundType.STONE).instabreak().lightLevel(s -> 15).noCollission().randomTicks()
-				.hasPostProcess((bs, br, bp) -> true).emissiveRendering((bs, br, bp) -> true));
+		super(BlockBehaviour.Properties.of(Material.BARRIER, MaterialColor.NONE).sound(SoundType.STONE).instabreak().lightLevel(s -> 15)
+				.noCollission().randomTicks().hasPostProcess((bs, br, bp) -> true).emissiveRendering((bs, br, bp) -> true));
```
Comment (PechayClub) [inline src/main/java/ch/teamburnuts/tnt_and_disparity/block/GlowBlock.java:32]: Updating the MCreator version lead to the change of the code.

File: src/main/java/ch/teamburnuts/tnt_and_disparity/entity/CarminiteLauncherCannonEntity.java [hunk 7]
```diff
@@ -97,17 +118,14 @@ public void tick() {
...
+		if (this.inGround)
 			this.discard();
-		}
 	}
 
 	public static CarminiteLauncherCannonEntity shoot(Level world, LivingEntity entity, Random random, float power, double damage, int knockback) {
 		CarminiteLauncherCannonEntity entityarrow = new CarminiteLauncherCannonEntity(TntAndDisparityModEntities.CARMINITE_LAUNCHER_CANNON, entity,
 				world);
-		entityarrow.shoot(entity.getLookAngle().x, entity.getLookAngle().y, entity.getLookAngle().z, power * 2, 0);
+		entityarrow.shoot(entity.getViewVector(1).x, entity.getViewVector(1).y, entity.getViewVector(1).z, power * 2, 0);
```
Comment (PechayClub) [inline src/main/java/ch/teamburnuts/tnt_and_disparity/entity/CarminiteLauncherCannonEntity.java:128]: Updating the MCreator version lead to the change of the code.

File: src/main/java/ch/teamburnuts/tnt_and_disparity/entity/DLKEntity.java [hunk 8]
```diff
@@ -72,7 +73,12 @@ public DLKEntity(EntityType<DLKEntity> type, Level world) {
 	@Override
 	protected void registerGoals() {
 		super.registerGoals();
-		this.goalSelector.addGoal(1, new MeleeAttackGoal(this, 1.2, false));
+		this.goalSelector.addGoal(1, new MeleeAttackGoal(this, 1.2, false) {
+			@Override
+			protected double getAttackReachSqr(LivingEntity entity) {
+				return (double) (4.0 + entity.getBbWidth() * entity.getBbWidth());
+			}
+		});
```
Comment (PechayClub) [inline src/main/java/ch/teamburnuts/tnt_and_disparity/entity/DLKEntity.java:81]: Updating the MCreator version lead to the change of the code.

File: src/main/java/ch/teamburnuts/tnt_and_disparity/entity/DLKKingEntity.java [hunk 9]
```diff
@@ -69,7 +70,12 @@ public DLKKingEntity(EntityType<DLKKingEntity> type, Level world) {
 	@Override
 	protected void registerGoals() {
 		super.registerGoals();
-		this.goalSelector.addGoal(1, new MeleeAttackGoal(this, 1.2, false));
+		this.goalSelector.addGoal(1, new MeleeAttackGoal(this, 1.2, false) {
+			@Override
+			protected double getAttackReachSqr(LivingEntity entity) {
+				return (double) (4.0 + entity.getBbWidth() * entity.getBbWidth());
+			}
+		});
```
Comment (PechayClub) [inline src/main/java/ch/teamburnuts/tnt_and_disparity/entity/DLKKingEntity.java:78]: Updating the MCreator version lead to the change of the code.

File: src/main/java/ch/teamburnuts/tnt_and_disparity/entity/EkorEntity.java [hunk 10]
```diff
@@ -66,7 +66,12 @@ public EkorEntity(EntityType<EkorEntity> type, Level world) {
 	@Override
 	protected void registerGoals() {
 		super.registerGoals();
-		this.goalSelector.addGoal(1, new MeleeAttackGoal(this, 1.2, false));
+		this.goalSelector.addGoal(1, new MeleeAttackGoal(this, 1.2, false) {
+			@Override
+			protected double getAttackReachSqr(LivingEntity entity) {
+				return (double) (4.0 + entity.getBbWidth() * entity.getBbWidth());
+			}
+		});
```
Comment (PechayClub) [inline src/main/java/ch/teamburnuts/tnt_and_disparity/entity/EkorEntity.java:74]: Updating the MCreator version lead to the change of the code.

File: src/main/java/ch/teamburnuts/tnt_and_disparity/entity/EkorShooterEntity.java [hunk 11]
```diff
@@ -99,14 +106,13 @@ public void tick() {
...
-		if (this.inGround) {
+		if (this.inGround)
 			this.discard();
-		}
 	}
 
 	public static EkorShooterEntity shoot(Level world, LivingEntity entity, Random random, float power, double damage, int knockback) {
 		EkorShooterEntity entityarrow = new EkorShooterEntity(TntAndDisparityModEntities.EKOR_PUPPET, entity, world);
-		entityarrow.shoot(entity.getLookAngle().x, entity.getLookAngle().y, entity.getLookAngle().z, power * 2, 0);
+		entityarrow.shoot(entity.getViewVector(1).x, entity.getViewVector(1).y, entity.getViewVector(1).z, power * 2, 0);
```
Comment (PechayClub) [inline src/main/java/ch/teamburnuts/tnt_and_disparity/entity/EkorShooterEntity.java:115]: Updating the MCreator version lead to the change of the code.

File: src/main/java/ch/teamburnuts/tnt_and_disparity/entity/GhostArmorEntity.java [hunk 12]
```diff
@@ -130,11 +130,16 @@ protected Vec3 getPosition() {
...
+			@Override
+			protected double getAttackReachSqr(LivingEntity entity) {
+				return (double) (4.0 + entity.getBbWidth() * entity.getBbWidth());
+			}
+		});
 		this.goalSelector.addGoal(5, new RandomLookAroundGoal(this));
 		this.goalSelector.addGoal(8, new MoveBackToVillageGoal(this, 0.6, false));
 		this.targetSelector.addGoal(9, new NearestAttackableTargetGoal(this, Player.class, false, false));
-		this.targetSelector.addGoal(10, new HurtByTargetGoal(this).setAlertOthers(this.getClass()));
+		this.targetSelector.addGoal(10, new HurtByTargetGoal(this).setAlertOthers());
```
Comment (PechayClub) [inline src/main/java/ch/teamburnuts/tnt_and_disparity/entity/GhostArmorEntity.java:142]: Updating the MCreator version lead to the change of the code.

File: src/main/java/ch/teamburnuts/tnt_and_disparity/entity/ShootableBBBEntity.java [hunk 13]
```diff
@@ -98,16 +119,13 @@ public void tick() {
...
-			RemoteBBBExplodeProcedure.execute(world, x, y, z);
+		if (this.inGround)
 			this.discard();
-		}
 	}
 
 	public static ShootableBBBEntity shoot(Level world, LivingEntity entity, Random random, float power, double damage, int knockback) {
 		ShootableBBBEntity entityarrow = new ShootableBBBEntity(TntAndDisparityModEntities.SHOOTABLE_BIG_BAD_BOOM, entity, world);
-		entityarrow.shoot(entity.getLookAngle().x, entity.getLookAngle().y, entity.getLookAngle().z, power * 2, 0);
+		entityarrow.shoot(entity.getViewVector(1).x, entity.getViewVector(1).y, entity.getViewVector(1).z, power * 2, 0);
```
Comment (PechayClub) [inline src/main/java/ch/teamburnuts/tnt_and_disparity/entity/ShootableBBBEntity.java:128]: Updating the MCreator version lead to the change of the code.

File: src/main/java/ch/teamburnuts/tnt_and_disparity/entity/StickmanEntity.java [hunk 14]
```diff
@@ -69,8 +70,13 @@ public StickmanEntity(EntityType<StickmanEntity> type, Level world) {
...
 		super.registerGoals();
-		this.goalSelector.addGoal(1, new MeleeAttackGoal(this, 3, true));
-		this.targetSelector.addGoal(2, new HurtByTargetGoal(this).setAlertOthers(this.getClass()));
+		this.goalSelector.addGoal(1, new MeleeAttackGoal(this, 3, true) {
+			@Override
+			protected double getAttackReachSqr(LivingEntity entity) {
+				return (double) (4.0 + entity.getBbWidth() * entity.getBbWidth());
+			}
+		});
+		this.targetSelector.addGoal(2, new HurtByTargetGoal(this).setAlertOthers());
```
Comment (PechayClub) [inline src/main/java/ch/teamburnuts/tnt_and_disparity/entity/StickmanEntity.java:79]: Updating the MCreator version lead to the change of the code.

File: src/main/java/ch/teamburnuts/tnt_and_disparity/entity/TeleportationRodEntity.java [hunk 15]
```diff
@@ -59,52 +61,62 @@ protected ItemStack getPickupItem() {
...
+		super.tick();
+		if (this.inGround)
 			this.discard();
-		}
 	}
 
 	public static TeleportationRodEntity shoot(Level world, LivingEntity entity, Random random, float power, double damage, int knockback) {
 		TeleportationRodEntity entityarrow = new TeleportationRodEntity(TntAndDisparityModEntities.TELEPORTATION_ROD, entity, world);
-		entityarrow.shoot(entity.getLookAngle().x, entity.getLookAngle().y, entity.getLookAngle().z, power * 2, 0);
+		entityarrow.shoot(entity.getViewVector(1).x, entity.getViewVector(1).y, entity.getViewVector(1).z, power * 2, 0);
```
Comment (PechayClub) [inline src/main/java/ch/teamburnuts/tnt_and_disparity/entity/TeleportationRodEntity.java:119]: Updating the MCreator version lead to the change of the code.

File: src/main/java/ch/teamburnuts/tnt_and_disparity/init/TntAndDisparityModEntities.java [hunk 16]
```diff
@@ -109,5 +114,6 @@ public static void registerAttributes(EntityAttributeCreationEvent event) {
 		event.put(CARMINITE_CANNON, CarminiteCannonEntity.createAttributes().build());
 		event.put(CARMINITE_CANNON_FLYING, CarminiteCannonFlyingEntity.createAttributes().build());
 		event.put(CARMINITE_CANNON_FLYING_FOR_PLAYER, CarminiteCannonFlyingForPlayerEntity.createAttributes().build());
+		event.put(LIVE_HOMING_ROCKET, LiveHomingRocketEntity.createAttributes().build());
```
Comment (PechayClub) [inline src/main/java/ch/teamburnuts/tnt_and_disparity/init/TntAndDisparityModEntities.java:117]: Updating the MCreator version lead to the change of the code.

**Review by Burnuts — APPROVED** [review]
It looks good.

File: src/main/java/ch/teamburnuts/tnt_and_disparity/init/TntAndDisparityModEntityRenderers.java [hunk 17]
```diff
@@ -37,5 +38,6 @@ public static void registerEntityRenderers(EntityRenderersEvent.RegisterRenderer
 		event.registerEntityRenderer(TntAndDisparityModEntities.CARMINITE_CANNON, CarminiteCannonRenderer::new);
 		event.registerEntityRenderer(TntAndDisparityModEntities.CARMINITE_CANNON_FLYING, CarminiteCannonFlyingRenderer::new);
 		event.registerEntityRenderer(TntAndDisparityModEntities.CARMINITE_CANNON_FLYING_FOR_PLAYER, CarminiteCannonFlyingForPlayerRenderer::new);
+		event.registerEntityRenderer(TntAndDisparityModEntities.LIVE_HOMING_ROCKET, LiveHomingRocketRenderer::new);
```
Comment (PechayClub) [inline src/main/java/ch/teamburnuts/tnt_and_disparity/init/TntAndDisparityModEntityRenderers.java:41]: Updating the MCreator version lead to the change of the code.

File: src/main/java/ch/teamburnuts/tnt_and_disparity/init/TntAndDisparityModItems.java [hunk 18]
```diff
@@ -90,6 +90,9 @@
 	public static final Item CARMINITE_RADIO_EMITTER = register(new CarminiteRadioEmitterItem());
 	public static final Item EMPTY = register(new EmptyItem());
 	public static final Item BOTTLE_OF_DISULFIDE = register(new BottleOfDisulfideItem());
+	public static final Item LIVE_HOMING_ROCKET = register(
+			new SpawnEggItem(TntAndDisparityModEntities.LIVE_HOMING_ROCKET, -1, -1, new Item.Properties().tab(CreativeModeTab.TAB_MISC))
+					.setRegistryName("live_homing_rocket_spawn_egg"));
```
Comment (PechayClub) [inline src/main/java/ch/teamburnuts/tnt_and_disparity/init/TntAndDisparityModItems.java:95]: Updating the MCreator version lead to the change of the code.

File: src/main/java/ch/teamburnuts/tnt_and_disparity/init/TntAndDisparityModModels.java [hunk 19]
```diff
@@ -26,6 +27,7 @@ public static void registerLayerDefinitions(EntityRenderersEvent.RegisterLayerDe
 		event.registerLayerDefinition(ModelEkorModel.LAYER_LOCATION, ModelEkorModel::createBodyLayer);
 		event.registerLayerDefinition(ModelCarminiteCannon.LAYER_LOCATION, ModelCarminiteCannon::createBodyLayer);
 		event.registerLayerDefinition(ModelCarminiteCannonFlying.LAYER_LOCATION, ModelCarminiteCannonFlying::createBodyLayer);
+		event.registerLayerDefinition(ModelLiveHomingRocket.LAYER_LOCATION, ModelLiveHomingRocket::createBodyLayer);
```
Comment (PechayClub) [inline src/main/java/ch/teamburnuts/tnt_and_disparity/init/TntAndDisparityModModels.java:30]: Updating the MCreator version lead to the change of the code.

File: src/main/java/ch/teamburnuts/tnt_and_disparity/item/BottleOfDisulfideItem.java [hunk 20]
```diff
@@ -26,7 +26,7 @@
 import ch.teamburnuts.tnt_and_disparity.procedures.DisulfideExplodeProcedure;
 import ch.teamburnuts.tnt_and_disparity.init.TntAndDisparityModItems;
 
-@Mod.EventBusSubscriber
+@Mod.EventBusSubscriber(bus = Mod.EventBusSubscriber.Bus.MOD)
```
Comment (PechayClub) [inline src/main/java/ch/teamburnuts/tnt_and_disparity/item/BottleOfDisulfideItem.java:29]: Updating the MCreator version lead to the change of the code.

File: src/main/java/ch/teamburnuts/tnt_and_disparity/world/dimension/DLKLandDimension.java [hunk 21]
```diff
@@ -67,7 +67,7 @@ public boolean isFoggyAt(int x, int y) {
 	}
 
 	@SubscribeEvent
-	public void onPlayerChangedDimensionEvent(PlayerEvent.PlayerChangedDimensionEvent event) {
+	public static void onPlayerChangedDimensionEvent(PlayerEvent.PlayerChangedDimensionEvent event) {
```
Comment (PechayClub) [inline src/main/java/ch/teamburnuts/tnt_and_disparity/world/dimension/DLKLandDimension.java:70]: Updating the MCreator version lead to the change of the code.

File: src/main/java/ch/teamburnuts/tnt_and_disparity/world/features/CarminiteBaseFeature.java [hunk 22]
```diff
@@ -1,69 +1,69 @@
...
+								.addProcessor(BlockIgnoreProcessor.STRUCTURE_AND_AIR).setIgnoreEntities(false),
+						context.random(), 2)) {
+					anyPlaced = true;
+				}
+			}
+			return anyPlaced;
+		}
+		return false;
+	}
+}
```
Comment (PechayClub) [inline src/main/java/ch/teamburnuts/tnt_and_disparity/world/features/CarminiteBaseFeature.java:69]: Updating the MCreator version lead to the change of the code.

File: src/main/resources/assets/tnt_and_disparity/lang/de_ch.json [hunk 23]
```diff
@@ -44,33 +46,33 @@
...
   "subtitles.dlkdlks": "DLK DLKs",
   "item.tnt_and_disparity.ekor_sound": "Ekor Klang",
-  "item.tnt_and_disparity.ghost_armor_spawn_egg": "Geisterrüstung Laich Ei",
+  "item.tnt_and_disparity.ghost_armor_spawn_egg": "GeisterrÃ¼stung Laich Ei",
   "block.tnt_and_disparity.glue_fence": "Leimzaun",
   "entity.tnt_and_disparity.carminite_cannon": "Carminit-Kanone",
-  "entity.tnt_and_disparity.dlk_king": "DLK König",
-  "item.tnt_and_disparity.big_bad_boom_resistant_armor_armor_boots": "Großer Schlechter Dröhnen Widerstandsfähige Rüstungsstiefel",
+  "entity.tnt_and_disparity.dlk_king": "DLK KÃ¶nig",
+  "item.tnt_and_disparity.big_bad_boom_resistant_armor_armor_boots": "GroÃŸer Schlechter DrÃ¶hnen WiderstandsfÃ¤hige RÃ¼stungsstiefel",
```
Comment (PechayClub) [inline src/main/resources/assets/tnt_and_disparity/lang/de_ch.json:75]: The languages got corrupted.

File: src/main/resources/assets/tnt_and_disparity/lang/en_us.json [hunk 24]
```diff
@@ -25,6 +26,7 @@
   "entity.tnt_and_disparity.ghost_armor": "Ghost Armor",
   "item.tnt_and_disparity.big_bad_boom_resistant_armor_armor_chestplate": "Big Bad Boom Resistant Armor Chestplate",
   "advancements.your_ears_explode.descr": "Get Ekored by Ekor",
+  "entity.tnt_and_disparity.live_homing_rocket": "Live Homing Rocket",
```
Comment (PechayClub) [inline src/main/resources/assets/tnt_and_disparity/lang/en_us.json:29]: Updating the MCreator version lead to the change of the code.

File: tnt_and_disparity.mcreator [hunk 25]
```diff
@@ -1567,77 +1607,79 @@
...
       "subtitles.dlkdlks": "DLK DLKs",
       "item.tnt_and_disparity.ekor_sound": "Ekor Klang",
-      "item.tnt_and_disparity.ghost_armor_spawn_egg": "Geisterr�stung Laich Ei",
+      "item.tnt_and_disparity.ghost_armor_spawn_egg": "GeisterrÃ¼stung Laich Ei",
       "block.tnt_and_disparity.glue_fence": "Leimzaun",
       "entity.tnt_and_disparity.carminite_cannon": "Carminit-Kanone",
-      "entity.tnt_and_disparity.dlk_king": "DLK K�nig",
-      "item.tnt_and_disparity.big_bad_boom_resistant_armor_armor_boots": "Gro�er Schlechter Dr�hnen Widerstandsf�hige R�stungsstiefel",
+      "entity.tnt_and_disparity.dlk_king": "DLK KÃ¶nig",
+      "item.tnt_and_disparity.big_bad_boom_resistant_armor_armor_boots": "GroÃŸer Schlechter DrÃ¶hnen WiderstandsfÃ¤hige RÃ¼stungsstiefel",
```
Comment (PechayClub) [inline tnt_and_disparity.mcreator:1682]: The languages got corrupted.

**Review by PechayClub — CHANGES_REQUESTED** [review]
The languages got corrupted. Please fix it.

File: src/main/resources/assets/tnt_and_disparity/lang/de_ch.json (hunk 23 above)
Comment (aaronschneider7797) [reply src/main/resources/assets/tnt_and_disparity/lang/de_ch.json:75]: Ok I will fix this.

File: tnt_and_disparity.mcreator (hunk 25 above)
Comment (aaronschneider7797) [reply tnt_and_disparity.mcreator]: Fixed in a new commit.

**Review by PechayClub — COMMENTED** [review]
I think it's fine.

**Pull request merged**